html/includes/common/top-devices.inc.php

```diff
@@ -17,51 +18,109 @@
 
 /**
- * Top devices by traffic
+ * Top devices
  * @author Sergiusz Paprzycki
  * @copyright 2015 Sergiusz Paprzycki <serek@walcz.net>
+ * @copyright 2016 Cercel Valentin <crc@nuamchefazi.ro>
  * @license GPL
  * @package LibreNMS
  * @subpackage Widgets
  */
 
+$sql = dbFetchRow('SELECT `settings` FROM `users_widgets` WHERE `user_id` = ? AND `widget_id` = ?', array($_SESSION["user_id"], '11'));
+$widget_mode = json_decode($sql['settings'], true);
+
+$top_query = $widget_mode['top_query'];
+
+$selected_traffic = '';
+$selected_uptime = '';
+$selected_ping = '';
+$selected_cpu = '';
+$selected_ram = '';
+
+switch ($top_query) {
+    case "0":
+        $selected_traffic = 'selected';
+        break;
+    case "1":
+        $selected_uptime = 'selected';
+        break;
+    case "2":
+        $selected_ping = 'selected';
+        break;
+    case "3":
+        $selected_cpu = 'selected';
+        break;
+    case "4":
+        $selected_ram = 'selected';
+        break;
+}
+
 if (defined('SHOW_SETTINGS') || empty($widget_settings)) {
     $common_output[] = '
-<form class="form" onsubmit="widget_settings(this); return false;">
-  <div class="form-group">
-    <div class="col-sm-6">
-      <label for="graph_type" class="control-label">Number of Devices: </label>
-    </div>
-    <div class="col-sm-4">
-      <input class="form-control" name="device_count" id="input_count_'.$unique_id.'" placeholder="ie. 5" value="'.$widget_settings['device_count'].'">
-    </div>
-  </div>
-  <div class="clearfix"></div>
-  <div class="form-group">
-    <div class="col-sm-6">
-      <label for="graph_type" class="control-label">Time interval (minutes): </label>
-    </div>
-    <div class="col-sm-4">
-      <input class="form-control" name="time_interval" id="input_time_'.$unique_id.'" placeholder="ie. 15" value="'.$widget_settings['time_interval'].'">
-    </div>
-  </div>
-  <div class="form-group">
-    <div class="col-sm-2">
-      <button type="submit" class="btn btn-default">Set</button>
-    </div>
-  </div>
-</form>
-    ';
+    <form class="form" onsubmit="widget_settings(this); return false;">
+        <div class="form-group">
+            <div class="col-sm-4">
+                <label for="title" class="control-label availability-map-widget-header">Widget title</label>
+            </div>
+            <div class="col-sm-6">
+                <input type="text" class="form-control" name="title" placeholder="Custom title for widget" value="' . htmlspecialchars($widget_settings['title']) . '">
+            </div>
+        </div>
+        
+        <div class="form-group">
+            <div class="col-sm-4">
+                <label for="top_query" class="control-label availability-map-widget-header">Top query</label>
+            </div>
+            <div class="col-sm-6">
+                <select class="form-control" name="top_query">
+                    <option value="0" ' . $selected_traffic . '>Traffic</option>
```

Comment:
I'd change this to be descriptive values so you don't have to visit this part of the code when checking what the switch call is doing.